sideboard/internal/imports.py

```diff
@@ -1,23 +1,32 @@
 from __future__ import unicode_literals
 import sys
 import importlib
+from collections import OrderedDict
 from glob import glob
+from itertools import chain
 from os.path import join, isdir, basename
 
 from sideboard.config import config
 
-plugins = {}
 
+plugins = OrderedDict()
+plugin_dirs = OrderedDict()
 
-def _discover_plugins():
-    ordered = list(reversed(config['priority_plugins']))
-    plugin_dirs = [d for d in glob(join(config['plugins_dir'], '*')) if isdir(d) and not basename(d).startswith('_')]
 
-    # glob() results are not ordered, so we sort here to ensure non-prioritized plugins load in the same order
-    # regardless of OS-dependent arbitrary ordering
-    plugin_dirs = sorted(plugin_dirs, key=lambda d: basename(d))
+def _discover_plugin_dirs():
```

Comment:
I split discovery of the plugin dirs into a separate function to make it easier to test.